bazel_tools/build_environment.bzl

```
@@ -1,6 +1,15 @@
 # Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+def is_figure(c):
+    return c == "0" or c == "1" or c == "2" or c == "3" or c == "4" or c == "5" or c == "6" or c == "7" or c == "8" or c == "9"
+
+def is_number(s):
+    for c in s.elems():
+        if not is_figure(c):
+            return False
+    return True
```

You can use `all` here.

```suggestion
def is_number(s):
    return all([is_figure(c) for c in s.elems()])
```

---

I was looking for `map` and was really sad not to find it. I did not even think of comprehension. I guess my Python is getting very rusty at this point.

---

Ah well. @cocreature's solution is much better. :grinning:

---

Actually I think I can one-up that.